src/Filter/ORM/AbstractFilter.php

```diff
@@ -8,19 +8,24 @@
 
 namespace Laminas\ApiTools\Doctrine\QueryBuilder\Filter\ORM;
 
-use DateTime;
 use Laminas\ApiTools\Doctrine\QueryBuilder\Filter\FilterInterface;
 use Laminas\ApiTools\Doctrine\QueryBuilder\Filter\Service\ORMFilterManager;
+use Laminas\ApiTools\Doctrine\QueryBuilder\Filter\TypeCastInterface;
 
 abstract class AbstractFilter implements FilterInterface
 {
     abstract public function filter($queryBuilder, $metadata, $option);
 
     protected $filterManager;
 
+    protected $typeCaster;
+
     public function __construct($params)
     {
         $this->setFilterManager($params[0]);
+        if (isset($params[1])) {
+            $this->setTypeCaster($params[1]);
+        }
     }
```

Comment:
I recognize this constructor existed before. What I'd do in this scenario is change the plugin manager to pass the following:

```php
[
    0 => $this,
    'filter_manager' => $this,
    'type_caster' => $this->resolveTypeCaster(),
]
```

and then here, you would do the following:

```php
public function __construct(array $params)
{

    $this->setFilterManager($this->extractFilterManagerFromConstructorParams($params));
    $this->setTypeCaster($params['type_caster'] ?? new TypeCaster());
}

private function extractFilterManagerFromConstructorParams(array $params)
{
    if (isset($params['filter_manager']) && $params['filter_manager'] instanceof ORMFilterManager) {
        return $params['filter_manager'];
    }
    if (isset($params[0]) && $params[0] instanceof ORMFilterManager) {
        return $params[0];
    }
    throw new InvalidArgumentException('Missing or invalid filter manager provided to ' . __CLASS__);
}
```

Reply:
:+1: Will do it exactly like that